src/ru/crazycoder/plugins/tabdir/SameFilenameTitleProvider.java

```diff
@@ -110,28 +107,36 @@ private String titleRelativeTo(final VirtualFile file, final FolderConfiguration
         String filePath = FileUtil.toSystemDependentName(file.getPath());
         String relativePath = FileUtil.getRelativePath(basePath, filePath, File.separatorChar);
         String[] parts = StringUtils.split(relativePath, File.separatorChar);
-        List<String> prefixes = new ArrayList<String>(Arrays.asList(parts));
-        if (prefixes.size() > 1) {
-            prefixes.remove(prefixes.size() - 1);
-            return TitleFormatter.format(prefixes, file.getPresentableName(), configuration);
-        } else {
-            return file.getPresentableName();
+        if (parts != null) {
+            List<String> prefixes = new ArrayList<>(Arrays.asList(parts));
+            if (prefixes.size() > 1) {
+                prefixes.remove(prefixes.size() - 1);
+                return TitleFormatter.format(prefixes, file.getPresentableName(), configuration);
+            }
+        }
+        GlobalConfig globalConfig = ApplicationManager.getApplication().getService(GlobalConfig.class);
+        String globalEmptyPrefix = (globalConfig.getEmptyPathReplacement() != null) ? globalConfig.getEmptyPathReplacement() : "";
+        String emptyPrefix = (configuration.getEmptyPathReplacement() != null) ? configuration.getEmptyPathReplacement() : "";
+        if (emptyPrefix.isBlank()) {
+            emptyPrefix = globalEmptyPrefix;
         }
+        return emptyPrefix + file.getPresentableName();
```

Comment:
Could you please describe(or point me) what is the purpose of `emptyPrefix`?
(to be honest, if it unit test can be provided - this is the best variant)

Reply:
So imagine your project has a structure like this
```
my-project/tests/test1.js
my-project/package.json
```

If you configure tabdir for a directory `my-project/`  with always show path relative to `my-project/` and for a directory `my-project/tests` with always show relative to `my-project/tests` and table title format `[{0}] {1}`, you will end up with an empty path (`[{0}]` part) for both `my-project/tests/test1.js` and `my-project/package.json` and you're not going to know which came from where - tab titles are only going to show `test1.js` and `package.json`.

If you configure empty file prefix for `my-project` as e.g. `[MP] ` and empty prefix for `my-project/tests` as e.g. `[T] `, you will end up with tab titles `[MP] package.json` and `[T] test1.js` instead, and you'll know which file came from which directory.

There are probably better examples for this, but I hope it makes sense.

Reply:
Got it. Case you've described is very similar to Scopes+File Colors builtin feature(but instead of prefix you can get custom tab colour for different folders).
It might be rare case, but since mostly all Tabdir users - are rare auditory, lets add this feature.

But I think this config in Global settings is unnecessary, can not imagine case where user want add some prefix to _all_ tabs in project. It should be available only in per-folder configuration. If you agree - new issue should be filled for future release to fix this.